packages/core/src/amazonqGumby/chat/session/session.ts

```diff
@@ -0,0 +1,15 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Messenger } from '../controller/messenger/messenger'
+
+export class Session {
+    // Used to keep track of whether or not the current session is currently authenticating/needs authenticating
+    public isAuthenticating: boolean = false
```

Comment:
How is this different than the other similar flags ?  E.g. https://github.com/aws/aws-toolkit-vscode/blob/46560e70b7dd29d83ff557b1db246db5091dd7fe/packages/core/src/amazonqFeatureDev/session/session.ts#L37

Can we use auth.ts or secondaryAuth.ts as the source of truth, instead of storing all of this shared state redundantly in many places? https://github.com/aws/aws-toolkit-vscode/blob/46560e70b7dd29d83ff557b1db246db5091dd7fe/packages/core/src/auth/auth.ts

Reply:
A gumby tab has no way of accessing a featureDev tab's session state. It only queries the global state through `authUtil.ts` (which in turn uses `auth.ts` and `secondaryAuth.ts`).  There may be a way to extract this to a common base class in the meantime though, which I'll puzzle over.